physical/raft/raft.go

```diff
@@ -1508,10 +1508,27 @@ func (b *RaftBackend) Transaction(ctx context.Context, txns []*physical.TxnEntry
 		return err
 	}
 
+	// Partition our txns into read and write. The read ones will get updated via the FSM to
+	// support undo logs. The write ones we process like normal.
+	readTxns := make([]*physical.TxnEntry, 0)
+	writeTxns := make([]*physical.TxnEntry, 0)
+
+	for _, txn := range txns {
+		if txn.Operation == physical.GetOperation {
+			readTxns = append(readTxns, txn)
+		} else {
+			writeTxns = append(writeTxns, txn)
+		}
+	}
+	err := b.fsm.Transaction(ctx, readTxns)
```

Comment:
Can we skip doing this if readTxns is empty?

Reply:
Sure